lib/rucio/tests/test_rse_protocol_ssh.py

```diff
@@ -0,0 +1,274 @@
+# -*- coding: utf-8 -*-
+# Copyright 2013-2021 CERN
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Authors:
+# - WeiJen Chang <e4523744@gmail.com>, 2013
+# - Ralph Vigne <ralph.vigne@cern.ch>, 2013
+# - Cheng-Hsi Chao <cheng-hsi.chao@cern.ch>, 2014
+# - Joaquín Bogado <jbogado@linti.unlp.edu.ar>, 2018
+# - Vincent Garonne <vincent.garonne@cern.ch>, 2018
+# - Benedikt Ziemons <benedikt.ziemons@cern.ch>, 2020
+# - Mayank Sharma <mayank.sharma@cern.ch>, 2021
+# - Radu Carpa <radu.carpa@cern.ch>, 2021
+
+from __future__ import print_function
+
+import os
+import shutil
+import tempfile
+import unittest
+from uuid import uuid4 as uuid
+
+import pytest
+
+from rucio.common import exception
+from rucio.common.utils import execute
+from rucio.rse import rsemanager
+from rucio.tests.common import skip_rse_tests_with_accounts, load_test_conf_file
+from rucio.tests.rsemgr_api_test import MgrTestCases
+
+
+@pytest.mark.noparallel(reason='creates and removes a test directory with a fixed name')
+@skip_rse_tests_with_accounts
+class TestRseSSH(unittest.TestCase):
+    tmpdir = None
+    user = None
+
+    @classmethod
+    def get_rse_info(cls):
+        """
+        Detects if containerized rses for ssh are available in the testing environment.
+        :return: A tuple (rse, prefix, hostname, port, user).
+        """
+        cmd = "rucio list-rses --expression 'test_container_ssh=True'"
```

Comment:
Another minor recommendation: It would be better IMO to use the RSEClient for fetching the SSH1 RSE, as done in the fixture referenced in the previous comment for XRD RSE's ;)